ObjectiveCloudantTests/QueryViewTests.m

```diff
@@ -0,0 +1,966 @@
+    op.key = @"key";
+    op.keys = @[ @"keys" ];
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [completionFailed fulfill];
+      XCTAssertNotNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+}
+
+- (void)testfailsMissingViewNAME
+{
+    XCTestExpectation *completionFailed = [self expectationWithDescription:@"Missing view name"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+    op.key = @"key";
+    op.keys = @[ @"keys" ];
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [completionFailed fulfill];
+      XCTAssertNotNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+}
+
+- (void)testCorrectlyQueryViewNoOptions
+{
+    XCTestExpectation *requestCompleted =
+        [self expectationWithDescription:@"Succesfully view query"];
+    XCTestExpectation *viewRowBlockCalled = [self expectationWithDescription:@"Document Found"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+
+    __block int numberOfDocs = 0;
+    __block BOOL firstDoc = YES;
+    op.viewRowBlock = ^(NSDictionary<NSString *, NSObject *> *doc) {
+      if (firstDoc) {
+          firstDoc = NO;
+          [viewRowBlockCalled fulfill];
+      }
+      numberOfDocs++;
+      XCTAssertNotNil(doc);
+      XCTAssertNotNil(doc[@"id"]);
+      XCTAssertNotNil(doc[@"key"]);
+      XCTAssertNotNil(doc[@"value"]);
+    };
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [requestCompleted fulfill];
+      XCTAssertNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+
+    XCTAssertEqual(10, numberOfDocs);
+}
+
+- (void)testCorrectlyQueryViewLimit1
+{
+    XCTestExpectation *requestCompleted =
+        [self expectationWithDescription:@"Succesfully view query"];
+    XCTestExpectation *viewRowBlockCalled = [self expectationWithDescription:@"Document Found"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+    op.limit = 1;
+
+    op.viewRowBlock = ^(NSDictionary<NSString *, NSObject *> *doc) {
+      [viewRowBlockCalled fulfill];
```

Comment:
Relying on behaviour like that makes the test more fragile -- someone refactoring it might change the test in a way that violates the behaviour without knowing they've done it.